compiler/testSuite/11-decimal/any1-v.bal

```diff
@@ -0,0 +1,40 @@
+import ballerina/io;
+
+type R record {|
+    any a;
+|};
+
+function bar(any a) returns any {
+    return a;
+}
+
+public function main() {
+    any a1 = 1d;
+    io:println(a1); // @output 1
+    any a2 = 1e6144d;
+    io:println(a2); // @output 1.000000000000000000000000000000000E+6144
```

Comment:
How is that right?

Reply:
I went through decNumber source code to find the reason behind that. Please go through the below table.

|decimal string|output of decQuadFromString|exponent from decQuadGetExponent|
|---|----|---|
|1E6110|1E+6110|6110|
|1E6111|1E+6111|6111|
|1E6112|1.0E+6112|6111|
|1E6113|1.00E+6113|6111|
|1E6114|1.000E+6114|6111|

- E_max for the adjusted exponent is `6144`
- When the exponent is adjusted, the coefficient is represented in scientific notation
- But with in 128 bits, the coefficient is a positive integer or zero
- Therefore when storing the exponent in 128 bits E_max should be adjusted as `6111 ( = 6144 - (34 - 1))`
- For example, consider 1E6113,
     - exponent is greater than 6111
     - in order to reduce the exponent to 6111, the coefficient should be changed as 100
     - now coefficient = 100, exponent = 6111
     - Therefore when printing we can see 1.00E+6113
Please check the source code https://github.com/dnotq/decNumber/blob/8852ef5b4339d611f493e94a791f3543b6ee81cd/decNumber-icu-368/decCommon.c#L417

Reply:
This is lot of technical details, but what is the bottom line? Is it a bug? is it because library doesn't guarantee (close to) shortest string?

Reply:
This is not a bug. Since the output of `toString` function is represented in scientific notation, we can see these trailing zeros. I have tried decQuadReduce method and it does not remove these trailing zeros. I will try GCC function and update.

Reply:
From what I understand you can't add trailing zeros to decimal because `1.0d === 1.00d`.

Reply:
This is a bit strange, but I think it is correct behavior because the exponent is limited to 6111.